zookeeper-jute/src/main/java/org/apache/jute/compiler/JCompType.java

```diff
@@ -54,7 +54,7 @@ String genJavaCompareTo(String fname) {
     }
 
     String genJavaEquals(String fname, String peer) {
-        return "    ret = " + fname + ".equals(" + peer + ");\n";
+        return "    ret = java.util.Objects.equals(" + fname + ", " + peer + ");\n";
```

Comment:
FYI [ZOOKEEPER-3521](https://issues.apache.org/jira/browse/ZOOKEEPER-3521). I don't think we are supposed to use `""` or `"<null>"` or any other placeholder in `ErrorTxn#path` but just fix ZOOKEEPER-3521.

Reply:
Unless we state that jute generated s will never contain a null field. Then we can think of `"<null>"` or `"<unknown>"` or sth. else.